virtual/vbox/vbox.go

```diff
@@ -503,3 +510,97 @@ func VBoxCmdContext(ctx context.Context, cmd string, cmds ...string) ([]byte, er
 
 	return out, nil
 }
+
+func CreateFileTransferRoot(path string) error {
+	FileTransferRoot = path
+	if _, err := os.Stat(path); !os.IsNotExist(err) {
+		//If path exists
+		log.Info().Str("transfer-root", path).Msg("File transfer root already exists... Continuing.")
+		return nil
+	}
+	log.Info().Str("transfer-root", path).Msg("File transfer root does not exists... Creating folder")
+	err := os.MkdirAll(path, 0777)
+	if err != nil {
+		log.Warn().Msgf("Error creating file transfer root: %s", err)
+		return err
+	}
+	err = os.Chmod(path, os.ModePerm)
+	if err != nil {
+		log.Warn().Msgf("Error setting folder perms on: %s error: %s", path, err)
+		return err
+	}
+	log.Info().Msg("File transfer root succesfully created!")
+	return nil
+}
+
+func CreateEventFolder(tag string) error {
```

Comment:
I am not so into how you have imagined the folders structure however, it is good practice to create folders temporarily if they are really temporary. 
Currently, there is no harm or at first glance I assume there is no harm to do permanent folders and removing them. However, it would be nice to create tempdirs, [tempdir](https://pkg.go.dev/testing#T.TempDir ). 
Again, just a proposal, nothing is must over here.